src/scenic/core/object_types.py

```diff
@@ -1117,22 +1117,44 @@ def distanceTo(self, point):
 
     @cached_method
     def intersects(self, other):
-        """Whether or not this object intersects another object"""
-        # For objects that are boxes and flat, we can take a fast route
-        if self._isPlanarBox and other._isPlanarBox:
+        """Whether or not this object intersects another object or region"""
+        ## Type Checking ##
+        if not isinstance(other, (Object, Region)):
```

Comment:
This won't work for distributions over objects or over regions, and I think it's possible to get into this function with those (it doesn't look like the veneer function calling this does any checks). We should either make this a `distributionFunction` or add some extra checks. (The former would be a little heavy-handed since it would trigger on objects with any random properties at all, even if they have static poses, but the latter would be more complex to implement.)

Reply:
Hmm good point. I'll add a fix.